processing/src/main/java/org/apache/druid/query/monomorphicprocessing/SpecializationService.java

```diff
@@ -209,14 +349,20 @@ T specialize(ImmutableMap<Class<?>, Class<?>> classRemapping)
     @SuppressWarnings("unchecked")
     private Class<T> defineClass(String specializedClassName, byte[] specializedClassBytecode)
     {
-      return (Class<T>) UNSAFE.defineClass(
-          specializedClassName,
-          specializedClassBytecode,
-          0,
-          specializedClassBytecode.length,
-          prototypeClass.getClassLoader(),
-          prototypeClass.getProtectionDomain()
-      );
+      if (DEFINE_CLASS_NOT_SUPPORTED_EXCEPTION != null) {
+        throw new UnsupportedOperationException(DEFINE_CLASS_NOT_SUPPORTED_EXCEPTION);
```

Comment:
Could you please add a message

Reply:
the message is already in the wrapped exception, this is only to capture the stack-track, similar to what we do here https://github.com/apache/incubator-druid/blob/master/core/src/main/java/org/apache/druid/java/util/common/ByteBufferUtils.java#L70